diff --git a/node_modules/packemon/cjs/Artifact.cjs b/Artifact.cjs.fix
index 497f1c7..02cd255 100644
--- a/node_modules/packemon/cjs/Artifact.cjs
+++ b/Artifact.cjs.fix
@@ -188,10 +188,13 @@ class Artifact {
       outputPath = inputPath;
     }
     const folder = format === 'lib' && this.sharedLib ? `lib/${this.platform}` : format;
+    // Folder path for declarations cannot have the platform subpath, since multiple builds can 
+		// share the same tsconfig
+		const declFolder = format;
     const entryExt = format === 'cjs' || format === 'mjs' ? format : 'js';
     let declExt;
     if (declaration) {
-      if (!inputFile || inputFile.endsWith('.ts')) {
+      if (!inputFile || /\.tsx?$/.test(inputFile)) {
         declExt = 'd.ts';
       } else if (inputFile.endsWith('.cts')) {
         declExt = 'd.cts';
@@ -201,7 +204,7 @@ class Artifact {
     }
     return {
       declExt,
-      declPath: declExt ? `./${new common.VirtualPath(folder, `${inputPath ?? outputPath}.${declExt}`)}` : undefined,
+      declPath: declExt ? `./${new common.VirtualPath(declFolder, `${inputPath ?? outputPath}.${declExt}`)}` : undefined,
       entryExt,
       entryPath: `./${new common.VirtualPath(folder, `${outputPath}.${entryExt}`)}`,
       folder
